tokio/src/future/maybe_done.rs

```diff
@@ -69,3 +70,54 @@ impl<Fut: Future> Future for MaybeDone<Fut> {
         Poll::Ready(())
     }
 }
+
+// Test for https://github.com/tokio-rs/tokio/issues/6729
+#[cfg(test)]
```

Comment:
should this also have `#[cfg(miri)]` on it? or do we also want these tests to run outside of miri?

Reply:
I mean, it doesn't really hurt to run it outside of miri too. :woman_shrugging:

Reply:
fair enough!